Remove save-analysis.

nnethercote opened the pull request:
Most tests involving save-analysis were removed, but I kept a few where the `-Zsave-analysis` was an add-on to the main thing being tested, rather than the main thing being tested.

Closes https://github.com/rust-lang/rust/issues/43606

nnethercote committed:
Remove save-analysis.

Most tests involving save-analysis were removed, but I kept a few where
the `-Zsave-analysis` was an add-on to the main thing being tested,
rather than the main thing being tested.

For `x.py install`, the `rust-analysis` target has been removed.

For `x.py dist`, the `rust-analysis` target has been kept in a
degenerate form: it just produces a single file `reduced.json`
indicating that save-analysis has been removed. This is necessary for
rustup to keep working.

Closes #43606.

nnethercote commented:
#43606 is the tracking issue for save analysis removal.

bjorn3 commented on src/test/ui/traits/alias/self-in-generics.rs:
```diff
@@ -1,9 +1,9 @@
+
+
+
+
```
Please remove extra whitespace.

nnethercote replied:
I did this in order to avoid changing line numbers in the `.stderr` files. But I can change them if people feel strongly about it.

JohnTitor replied:
I'd prefer to bless stderr files as it's quite unclear for future contributors why these tests have extra lines.

bjorn3 commented on src/test/ui/type-alias-impl-trait/issue-63279.rs:
```diff
@@ -1,7 +1,7 @@
 #![feature(type_alias_impl_trait)]
 
+
+
```
Same

lqd commented on src/test/ui/issues/issue-3763.rs:
```diff
@@ -1,5 +1,5 @@
-// compile-flags: -Zsave-analysis
-// Also regression test for #69416
+// Regression test for #3763
```
69416 was an ICE for save-analysis on a test for issue 3763

nnethercote replied:
Yes. See the filename. This test was added for #3763 originally and then later `-Zsave-analysis` was added in #69416.

JohnTitor commented on src/test/ui/feature-gates/feature-gate-associated_type_bounds.rs:
```diff
@@ -1,8 +1,8 @@
-// compile-flags: -Zsave-analysis
-// This is also a regression test for #69415 and the above flag is needed.
```
This test is for save-analysis and could also be removed.

nnethercote replied:
I tried that and got a tidy error about a missing test for the `associated_type_bound` feature gate.

JohnTitor replied:
Ah hm, that makes sense then. Thanks for clarifying!

JohnTitor commented on src/test/ui/asm/issue-72570.rs:
```diff
@@ -1,6 +1,6 @@
-// compile-flags: -Zsave-analysis
 // needs-asm-support
-// Also test for #72960
+
+
```
These are unnecessary too.

lqd commented on src/test/ui/methods/assign-to-method.rs:
```diff
@@ -1,24 +0,0 @@
-// compile-flags: -Zsave-analysis
```
This test however existed before save-analysis and was repurposed to test a save-analysis ICE at the same time. We can just remove the compile flag and keep the test.

nnethercote commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion.

@rustbot label: +S-waiting-on-perf

bors commented:
:hourglass: Trying commit f4d60e6cf969a8262f3fac33eb243c1294ff54cb with merge 19e674d3ec94f9dd85de3348e63ebb4baa3d91df...
<!-- homu: {"type":"TryBuildStarted","head_sha":"f4d60e6cf969a8262f3fac33eb243c1294ff54cb","merge_sha":"19e674d3ec94f9dd85de3348e63ebb4baa3d91df"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/3059624842/jobs/4938643198)
Build commit: 19e674d3ec94f9dd85de3348e63ebb4baa3d91df (`19e674d3ec94f9dd85de3348e63ebb4baa3d91df`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/3059624842/jobs/4938643198"},"merge_sha":"19e674d3ec94f9dd85de3348e63ebb4baa3d91df"} -->

rust-timer commented:
Queued 19e674d3ec94f9dd85de3348e63ebb4baa3d91df with parent 0a27ac16110c0a546504a454c9891d132886f453, future [comparison URL](https://perf.rust-lang.org/compare.html?start=0a27ac16110c0a546504a454c9891d132886f453&end=19e674d3ec94f9dd85de3348e63ebb4baa3d91df).

est31 commented:
Would it be possible to wait with merging of this, maybe 2-3 weeks? Due to this PR, I've switched cargo-udeps over to use binary depinfo, and made the 0.1.33 release, but it needs some time until the update arrives at users. On the other hand cargo-udeps has to be used with nightly rustc, so a removal would get to them very quickly.

lqd commented:
@est31 there's a bit more work to be done here related to rustup, and will need FCP I think (which lasts 10 at least days or so IIRC), so there's time yeah.

nnethercote commented:
The perf run shows that bootstrap time was reduced by about 4.6 seconds due to the removal of the `rust_save_analysis` crate.

nnethercote commented:
I addressed all the comments about tests.

nnethercote commented:
@rfcbot fcp

nnethercote commented:
@rfcbot fcp merge

bors commented:
:umbrella: The latest upstream changes (presumably #101938) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

eddyb commented:
> #43606 is the tracking issue for save analysis removal.

Not super important, but if you want you could add "Closes #43606" to the PR description.

nikomatsakis commented:
@rfcbot revewied

nnethercote commented:
> Once a majority of reviewers approve (and at most 2 approvals are outstanding)

With 14 compiler team members, I think this means 12 must approve. We currently have 10 approvals.  (But I just was doing some experimental clicking and it's possible that I may have undone a vote from @estebank, sorry!)

The PR probably can't land in its current form, because something more will be needed for `rustup` to be happy, e.g. leaving behind an empty save-analysis component. I propose waiting until the 12 approvals are in before starting that work, though I'm happy to hear counter-suggestions.

nnethercote commented:
r? @Mark-Simulacrum 

Most of the changes are straightforward. The bootstrap changes are the trickiest. I removed the `x install` support for `rust-analysis` and changed the `x dist` support so it produces stub output. I think I got it right, but I don't understand that code very well.

tmandry commented:
Unfortunately Google's [indexing tool](https://github.com/kythe/kythe) depends on save-analysis, and we haven't yet been able to find anyone with time to fix it.

Lately we've investigated using the rust-analyzer tooling for Fuchsia, but all three of rust-analyzer's CLI tools report many panics and then loop infinitely on Fuchsia's codebase, with varying backtraces, in a way that is not easy for us to diagnose or minimize.

Using save-analysis always came with a risk of it being removed, but at the time of developing that tooling, rust-analyzer was not yet in a place where it could be used for this purpose. It seems like there are still some issues we have that need to be chased down. So while I support switching to rust-analyzer (and ultimately removing save-analysis), this does not appear to be as straightforward as I hoped.

In the meantime, if save-analysis is removed we will lose support for the Rust code indexing on https://cs.opensource.google/ and related websites. What's the impact of that? It's not astronomical, but not trivial either. Many developers use the code navigation for getting up to speed in Google's open source code bases that use Rust.

Personally I would much prefer it if we had a longer warning period prior to removal to work out the issues that keep us from adopting rust-analyzer. An announcement e.g. on the Inside Rust blog might also help spread the message to other users who depend on it.

bors commented:
:umbrella: The latest upstream changes (presumably #103345) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

wesleywiser commented:
Nominating for compiler team discussion. We need to decide when we want to land this change. There's [been an ask on the Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Future.20of.20save-analysis/near/304896312) for us to wait 3 months so that tooling which currently depends on save-analysis can be rewritten to use r-a instead. We should also discuss how much messaging we need to do prior to merging.

apiraino commented:
Visited during T-compiler meeting, conversation will follow on this [Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Future.20of.20save-analysis) (also with regard to the question if save-analysis can wait 3 months before being removed, which seems fine).

@rustbot label -I-compiler-nominated

nnethercote commented:
Three months from today will be February 14, 2023. I will merge this PR on that date or shortly after. That will be one day short of five months after this PR was filed. Thanks to everyone for the discussions so far.

nnethercote commented:
@Mark-Simulacrum: I have updated. We're getting close the Feb 14 trigger date, should be a good time to review. The tricky part here is the bootstrap stuff and its interaction with rustup, I think.

Mark-Simulacrum reviewed (approved):
Code wise this seems fine to me, I think we can move ahead on Tuesday (?).

I expect it may bounce a few times in CI so feel free to re-approve with minor changes. (r=me)

nnethercote commented:
@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit c3d75baf3314896d79bc100387702218e3699c5d has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum c3d75baf3314896d79bc100387702218e3699c5d -->
<!-- homu: {"type":"Approved","sha":"c3d75baf3314896d79bc100387702218e3699c5d","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit c3d75baf3314896d79bc100387702218e3699c5d with merge 82f6e1f1fb22394341ed66f1b1f9bebca505f440...
<!-- homu: {"type":"BuildStarted","head_sha":"c3d75baf3314896d79bc100387702218e3699c5d","merge_sha":"82f6e1f1fb22394341ed66f1b1f9bebca505f440"} -->

bors commented:
:umbrella: The latest upstream changes (presumably #108079) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4183637397/jobs/7252698217)
Approved by: Mark-Simulacrum
Pushing 82f6e1f1fb22394341ed66f1b1f9bebca505f440 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4183637397/jobs/7252698217"},"merge_sha":"82f6e1f1fb22394341ed66f1b1f9bebca505f440"} -->

rust-timer commented:
Finished benchmarking commit (82f6e1f1fb22394341ed66f1b1f9bebca505f440): [comparison URL](https://perf.rust-lang.org/compare.html?start=999ac5f7770bff68bd65f490990d32c3ec1faaa6&end=82f6e1f1fb22394341ed66f1b1f9bebca505f440&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=999ac5f7770bff68bd65f490990d32c3ec1faaa6&end=82f6e1f1fb22394341ed66f1b1f9bebca505f440&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=999ac5f7770bff68bd65f490990d32c3ec1faaa6&end=82f6e1f1fb22394341ed66f1b1f9bebca505f440&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -1.2% | [-1.2%, -1.2%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=999ac5f7770bff68bd65f490990d32c3ec1faaa6&end=82f6e1f1fb22394341ed66f1b1f9bebca505f440&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

bors commented:
:lock: Merge conflict

This pull request and the master branch diverged in a way that cannot be automatically merged. Please rebase on top of the latest master branch, and let the reviewer approve again.

<details><summary>How do I rebase?</summary>

Assuming `self` is your fork and `upstream` is this repository, you can resolve the conflict following these steps:

1. `git checkout rm-save-analysis` *(switch to your branch)*
2. `git fetch upstream master` *(retrieve the latest master)*
3. `git rebase upstream/master -p` *(rebase on top of it)*
4. Follow the on-screen instruction to resolve conflicts (check `git status` if you got lost).
5. `git push self rm-save-analysis --force-with-lease` *(update this PR)*

You may also read [*Git Rebasing to Resolve Conflicts* by Drew Blessing](http://blessing.io/git/git-rebase/open-source/2015/08/23/git-rebasing-to-resolve-conflicts.html) for a short tutorial.

Please avoid the ["**Resolve conflicts**" button](https://help.github.com/articles/resolving-a-merge-conflict-on-github/) on GitHub. It uses `git merge` instead of `git rebase` which makes the PR commit history more difficult to read.

Sometimes step 4 will complete without asking for resolution. This is usually due to difference between how `Cargo.lock` conflict is handled during merge and rebase. This is normal, and you should still perform step 5 to update this PR.

</details>

<details><summary>Error message</summary>

```text
Removing tests/ui/save-analysis/issue-89066.stderr
Removing tests/ui/save-analysis/issue-89066.rs
Removing tests/ui/save-analysis/issue-73022.rs
Removing tests/ui/save-analysis/issue-73020.stderr
Removing tests/ui/save-analysis/issue-73020.rs
Removing tests/ui/save-analysis/issue-72267.stderr
Removing tests/ui/save-analysis/issue-72267.rs
Removing tests/ui/save-analysis/issue-68621.stderr
Removing tests/ui/save-analysis/issue-68621.rs
Removing tests/ui/save-analysis/issue-65590.rs
Removing tests/ui/save-analysis/issue-65411.rs
Removing tests/ui/save-analysis/issue-64659.rs
Removing tests/ui/save-analysis/issue-63663.rs
Removing tests/ui/save-analysis/issue-59134-1.stderr
Removing tests/ui/save-analysis/issue-59134-1.rs
Removing tests/ui/save-analysis/issue-59134-0.stderr
Removing tests/ui/save-analysis/issue-59134-0.rs
Removing tests/ui/save-analysis/issue-37323.rs
Removing tests/ui/save-analysis/issue-26459.stderr
Removing tests/ui/save-analysis/issue-26459.rs
Removing tests/ui/save-analysis/emit-notifications.stderr
Removing tests/ui/save-analysis/emit-notifications.rs
Removing tests/ui/save-analysis/emit-notifications.polonius.stderr
Auto-merging tests/rustdoc-ui/z-help.stdout
Removing tests/run-make-fulldeps/save-analysis/krate2.rs
Removing tests/run-make-fulldeps/save-analysis/foo.rs
Removing tests/run-make-fulldeps/save-analysis/extra-docs.md
Removing tests/run-make-fulldeps/save-analysis/SubDir/mod.rs
Removing tests/run-make-fulldeps/save-analysis/SameDir3.rs
Removing tests/run-make-fulldeps/save-analysis/SameDir.rs
Removing tests/run-make-fulldeps/save-analysis/Makefile
Removing tests/run-make-fulldeps/save-analysis-rfc2126/validate_json.py
Removing tests/run-make-fulldeps/save-analysis-rfc2126/krate2.rs
Removing tests/run-make-fulldeps/save-analysis-rfc2126/extern_absolute_paths.rs
Removing tests/run-make-fulldeps/save-analysis-rfc2126/Makefile
Removing tests/run-make-fulldeps/save-analysis-fail/krate2.rs
Removing tests/run-make-fulldeps/save-analysis-fail/foo.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SubDir/mod.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SameDir3.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SameDir.rs
Removing tests/run-make-fulldeps/save-analysis-fail/Makefile
Removing tests/run-make-fulldeps/issues-41478-43796/a.rs
Removing tests/run-make-fulldeps/issues-41478-43796/Makefile
Auto-merging src/tools/tidy/src/deps.rs
Auto-merging compiler/rustc_session/src/options.rs
Removing compiler/rustc_save_analysis/src/span_utils.rs
Removing compiler/rustc_save_analysis/src/sig.rs
CONFLICT (modify/delete): compiler/rustc_save_analysis/src/lib.rs deleted in heads/homu-tmp and modified in HEAD. Version HEAD of compiler/rustc_save_analysis/src/lib.rs left in tree.
Removing compiler/rustc_save_analysis/src/errors.rs
Removing compiler/rustc_save_analysis/src/dumper.rs
Removing compiler/rustc_save_analysis/src/dump_visitor.rs
Removing compiler/rustc_save_analysis/Cargo.toml
Auto-merging compiler/rustc_interface/src/tests.rs
Removing compiler/rustc_error_messages/locales/en-US/save_analysis.ftl
Auto-merging compiler/rustc_driver_impl/src/lib.rs
Auto-merging Cargo.lock
Automatic merge failed; fix conflicts and then commit the result.

```

</details>

bors commented:
:umbrella: The latest upstream changes (presumably #108006) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

nnethercote commented:
I rebased.

@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit 2cc20a37160789d20f372a978cfdbd4a46ac485b has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 2cc20a37160789d20f372a978cfdbd4a46ac485b -->
<!-- homu: {"type":"Approved","sha":"2cc20a37160789d20f372a978cfdbd4a46ac485b","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:lock: Merge conflict

This pull request and the master branch diverged in a way that cannot be automatically merged. Please rebase on top of the latest master branch, and let the reviewer approve again.

<details><summary>How do I rebase?</summary>

Assuming `self` is your fork and `upstream` is this repository, you can resolve the conflict following these steps:

1. `git checkout rm-save-analysis` *(switch to your branch)*
2. `git fetch upstream master` *(retrieve the latest master)*
3. `git rebase upstream/master -p` *(rebase on top of it)*
4. Follow the on-screen instruction to resolve conflicts (check `git status` if you got lost).
5. `git push self rm-save-analysis --force-with-lease` *(update this PR)*

You may also read [*Git Rebasing to Resolve Conflicts* by Drew Blessing](http://blessing.io/git/git-rebase/open-source/2015/08/23/git-rebasing-to-resolve-conflicts.html) for a short tutorial.

Please avoid the ["**Resolve conflicts**" button](https://help.github.com/articles/resolving-a-merge-conflict-on-github/) on GitHub. It uses `git merge` instead of `git rebase` which makes the PR commit history more difficult to read.

Sometimes step 4 will complete without asking for resolution. This is usually due to difference between how `Cargo.lock` conflict is handled during merge and rebase. This is normal, and you should still perform step 5 to update this PR.

</details>

<details><summary>Error message</summary>

```text
Removing tests/ui/save-analysis/issue-89066.stderr
Removing tests/ui/save-analysis/issue-89066.rs
Removing tests/ui/save-analysis/issue-73022.rs
Removing tests/ui/save-analysis/issue-73020.stderr
Removing tests/ui/save-analysis/issue-73020.rs
Removing tests/ui/save-analysis/issue-72267.stderr
Removing tests/ui/save-analysis/issue-72267.rs
Removing tests/ui/save-analysis/issue-68621.stderr
Removing tests/ui/save-analysis/issue-68621.rs
Removing tests/ui/save-analysis/issue-65590.rs
Removing tests/ui/save-analysis/issue-65411.rs
Removing tests/ui/save-analysis/issue-64659.rs
Removing tests/ui/save-analysis/issue-63663.rs
Removing tests/ui/save-analysis/issue-59134-1.stderr
Removing tests/ui/save-analysis/issue-59134-1.rs
Removing tests/ui/save-analysis/issue-59134-0.stderr
Removing tests/ui/save-analysis/issue-59134-0.rs
Removing tests/ui/save-analysis/issue-37323.rs
Removing tests/ui/save-analysis/issue-26459.stderr
Removing tests/ui/save-analysis/issue-26459.rs
Removing tests/ui/save-analysis/emit-notifications.stderr
Removing tests/ui/save-analysis/emit-notifications.rs
Removing tests/ui/save-analysis/emit-notifications.polonius.stderr
Auto-merging tests/ui/feature-gates/feature-gate-associated_type_bounds.stderr
CONFLICT (content): Merge conflict in tests/ui/feature-gates/feature-gate-associated_type_bounds.stderr
Removing tests/run-make-fulldeps/save-analysis/krate2.rs
Removing tests/run-make-fulldeps/save-analysis/foo.rs
Removing tests/run-make-fulldeps/save-analysis/extra-docs.md
Removing tests/run-make-fulldeps/save-analysis/SubDir/mod.rs
Removing tests/run-make-fulldeps/save-analysis/SameDir3.rs
Removing tests/run-make-fulldeps/save-analysis/SameDir.rs
Removing tests/run-make-fulldeps/save-analysis/Makefile
Removing tests/run-make-fulldeps/save-analysis-rfc2126/validate_json.py
Removing tests/run-make-fulldeps/save-analysis-rfc2126/krate2.rs
Removing tests/run-make-fulldeps/save-analysis-rfc2126/extern_absolute_paths.rs
Removing tests/run-make-fulldeps/save-analysis-rfc2126/Makefile
Removing tests/run-make-fulldeps/save-analysis-fail/krate2.rs
Removing tests/run-make-fulldeps/save-analysis-fail/foo.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SubDir/mod.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SameDir3.rs
Removing tests/run-make-fulldeps/save-analysis-fail/SameDir.rs
Removing tests/run-make-fulldeps/save-analysis-fail/Makefile
Removing tests/run-make-fulldeps/issues-41478-43796/a.rs
Removing tests/run-make-fulldeps/issues-41478-43796/Makefile
Removing compiler/rustc_save_analysis/src/span_utils.rs
Removing compiler/rustc_save_analysis/src/sig.rs
Removing compiler/rustc_save_analysis/src/lib.rs
Removing compiler/rustc_save_analysis/src/errors.rs
Removing compiler/rustc_save_analysis/src/dumper.rs
Removing compiler/rustc_save_analysis/src/dump_visitor.rs
Removing compiler/rustc_save_analysis/Cargo.toml
Removing compiler/rustc_error_messages/locales/en-US/save_analysis.ftl
Automatic merge failed; fix conflicts and then commit the result.

```

</details>

bors commented:
:umbrella: The latest upstream changes (presumably #108096) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

nnethercote commented:
I rebased again.

@bors r=Mark-Simulacrum

bors commented:
:pushpin: Commit 22a5125a367eedc9f1d6b49bac6557a4b4864423 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 22a5125a367eedc9f1d6b49bac6557a4b4864423 -->
<!-- homu: {"type":"Approved","sha":"22a5125a367eedc9f1d6b49bac6557a4b4864423","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:hourglass: Testing commit 22a5125a367eedc9f1d6b49bac6557a4b4864423 with merge af3c8b27266e290cf65704284f6862d0f90ee4fc...
<!-- homu: {"type":"BuildStarted","head_sha":"22a5125a367eedc9f1d6b49bac6557a4b4864423","merge_sha":"af3c8b27266e290cf65704284f6862d0f90ee4fc"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4193087674/jobs/7273366512)
Approved by: Mark-Simulacrum
Pushing af3c8b27266e290cf65704284f6862d0f90ee4fc to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4193087674/jobs/7273366512"},"merge_sha":"af3c8b27266e290cf65704284f6862d0f90ee4fc"} -->

rust-timer commented:
Finished benchmarking commit (af3c8b27266e290cf65704284f6862d0f90ee4fc): [comparison URL](https://perf.rust-lang.org/compare.html?start=4b34c7b766a3d64d3f2a9bff06d0e53648a11013&end=af3c8b27266e290cf65704284f6862d0f90ee4fc&stat=instructions:u).

### Overall result: ✅ improvements - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=4b34c7b766a3d64d3f2a9bff06d0e53648a11013&end=af3c8b27266e290cf65704284f6862d0f90ee4fc&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.6% | [-2.6%, -2.6%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=4b34c7b766a3d64d3f2a9bff06d0e53648a11013&end=af3c8b27266e290cf65704284f6862d0f90ee4fc&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | 4.7%  | [4.7%, 4.7%]   | 1     |
| Regressions ❌ <br /> (secondary)  | 1.8%  | [1.0%, 2.5%]   | 2     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -1.0% | [-1.0%, -1.0%] | 1     |
| All ❌✅ (primary)                 | 4.7%  | [4.7%, 4.7%]   | 1     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=4b34c7b766a3d64d3f2a9bff06d0e53648a11013&end=af3c8b27266e290cf65704284f6862d0f90ee4fc&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
